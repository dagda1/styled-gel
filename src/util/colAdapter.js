import { breakpoints, defaultColumns } from '../components/constants';
import { findLastIndex } from 'lodash';

export const colAdapter = props => {
  const noOfColumns = (props.theme && props.theme.columns) || defaultColumns;

  const width = val => Number(val) / noOfColumns;

  const widths = Object.keys(breakpoints).map(breakPoint => {
    const value = props[breakPoint];

    if (!value) return null;

    return width(value);
  });

  if (props.xs) {
    widths.unshift(width(props.xs));
  }

  const endIndex = findLastIndex(widths, x => !!x);

  if (endIndex === -1) {
    return { w: 1 };
  }

  const w = [];

  for (var i = 0; i <= endIndex; i++) {
    w.push(widths[i]);
  }

  return { w };
};
