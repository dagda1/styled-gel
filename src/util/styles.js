import { css } from 'styled-components';
import { breakPoints } from '../components';

export const media = Object.keys(breakPoints)
  .reduce((acc, key) => {
      acc[key] = (...args) => css`
      @media (min-width: ${breakPoints[key]}) {
        ${ css(...args) }
      }
    `;

    return acc;
  }, {});
