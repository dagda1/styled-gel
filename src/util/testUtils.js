export const stripComments = str =>
  str.replace(/\/\*.*?\*\/\n?/g, '')

export const stripWhitespace = str =>
  str.trim().replace(/([;{}])/g, '$1  ').replace(/\s+/g, ' ')

export const getCSSFromDom = () =>
  Array.from(document.querySelectorAll('style'))
      .map(tag => tag.innerHTML)
      .join('\n')
      .replace(/ {/g, '{')
      .replace(/:\s+;/g, ':;')
      .split(';')
      .map(stripComments)
      .map(stripWhitespace)
      .filter(x => x.trim('').length);
