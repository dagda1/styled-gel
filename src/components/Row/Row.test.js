import React from 'react';
import { shallow } from 'enzyme';
import { Row, Col } from '..';
import { getCSSFromDom } from '../../util/testUtils';
import renderer from 'react-test-renderer';

const wrap = (props = {}) => shallow(<Row {...props} />).dive();

it('should render a div with display flexbox and flex-wrap', () => {
  wrap();

  const css = getCSSFromDom();

  expect(css.includes('display: flex')).toBe(true);
  expect(css.includes('flex-wrap: wrap')).toBe(true);
});

it('should remove gutter with flush prop', () => {
  wrap({ flush: true });

  const css = getCSSFromDom();

  expect(css.includes('margin-left: 0'));
  expect(css.includes('margin-right: 0'));
});

it('should pass flush property to children', () => {
  const wrapper = wrap({ flush: true, children: <Col>Foo Bar</Col> });

  const col = wrapper.find(Col);

  expect(col.props().flush).toBe(true);
});

it('renders correctly', () => {
  const row = renderer.create(<Row>Foo</Row>);

  expect(row).toMatchSnapshot();
});