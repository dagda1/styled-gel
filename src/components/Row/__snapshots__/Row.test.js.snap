// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-htpNat gifSVp"
  wrap={true}
>
  Foo
</div>
`;
